Rubberduck.Parsing/ComReflection/ComParameter.cs

```diff
@@ -87,48 +93,60 @@ private void GetParameterType(TYPEDESC desc, ITypeInfo info)
+                }
+
+                try
+                {
+                    info.GetRefTypeInfo(href, out ITypeInfo refTypeInfo);
+                    refTypeInfo.GetTypeAttr(out IntPtr attribPtr);
+                    using (DisposalActionContainer.Create(attribPtr, refTypeInfo.ReleaseTypeAttr))
                     {
-                        info.GetRefTypeInfo(href, out ITypeInfo refTypeInfo);
-                        refTypeInfo.GetTypeAttr(out IntPtr attribPtr);
-                        using (DisposalActionContainer.Create(attribPtr, refTypeInfo.ReleaseTypeAttr))
+                        var attribs = Marshal.PtrToStructure<TYPEATTR>(attribPtr);
+                        var type = new ComDocumentation(refTypeInfo, -1).Name;
+                        if (attribs.typekind == TYPEKIND.TKIND_ENUM)
                         {
-                            var attribs = Marshal.PtrToStructure<TYPEATTR>(attribPtr);
-                            if (attribs.typekind == TYPEKIND.TKIND_ENUM)
-                            {
-                                _enumGuid = attribs.guid;
-                            }
-                            _type = new ComDocumentation(refTypeInfo, -1).Name;
+                            _typeName = new ComTypeName(Project, type, attribs.guid, Guid.Empty);
+                        }
+                        else if (attribs.typekind == TYPEKIND.TKIND_ALIAS)
+                        {
+                            _typeName = new ComTypeName(Project, type, Guid.Empty, attribs.guid);
+                        }
+                        else
+                        {
+                            _typeName = new ComTypeName(Project, type);
                         }
                     }
-                    catch (COMException) { }
-                    break;
-                case VarEnum.VT_SAFEARRAY:
-                case VarEnum.VT_CARRAY:
-                case VarEnum.VT_ARRAY:
-                    tdesc = (TYPEDESC)Marshal.PtrToStructure(desc.lpValue, typeof(TYPEDESC));
-                    GetParameterType(tdesc, info);
-                    IsArray = true;
-                    break;
-                default:
-                    if (ComVariant.TypeNames.TryGetValue(vt, out string result))
-                    {
-                        _type = result;
-                    }
-                    break;
+                }
+                catch (COMException)
+                {
+                    _typeName = new ComTypeName(Project, "Object");
+                }
+            }
+            else if (vt == VarEnum.VT_SAFEARRAY || vt == VarEnum.VT_CARRAY || vt.HasFlag(VarEnum.VT_ARRAY))
+            {
+                tdesc = Marshal.PtrToStructure<TYPEDESC>(desc.lpValue);
+                GetParameterType(tdesc, info);
+                IsArray = true;
+            }
+            else if (vt == VarEnum.VT_HRESULT)
+            {
+                _typeName = new ComTypeName(Project, "Long");
```

Comment:
`Tokens.Long`? Also, I think a comment is needed here as to why we are making it a long (I'm stumbling primarily because I don't expect VB to ever show a `HRESULT`)

Reply:
Above re the token (fixed). The MSXML6 library has a hidden interface that takes an `HRESULT` as a freaking _parameter_, so it has to be handled. Who knew?